src/main/java/se/iths/librarysystem/service/BookService.java

```diff
@@ -24,4 +27,11 @@ public BookEntity createBook(BookEntity bookEntity) {
     public Iterable<BookEntity> getBooksByIsbn(String isbn) {
         return bookRepository.findByIsbn(isbn);
     }
+
+    public Optional<BookEntity> findBookById(Long id){return bookRepository.findById(id);}
+
+    public void deleteBook(Long id){
+        BookEntity foundItem = bookRepository.findById(id).orElseThrow(EntityNotFoundException::new);
```

Comment:
We need to add exception handling for `EntityNotFoundException` to  `GlobalExceptionHandler` so that the user receives a meaningful status instead of 500 - Server Failure. Alternatively we could use our custom `IdNotFoundException` which already has exception handling.